packages/core/src/dispatcher.ts

```diff
@@ -387,6 +387,8 @@ export class Dispatcher {
 
           if (retries > 0) {
             continue
+          } else {
+            throw new Error(`Timeout error while loading CID ${asCid.toString()} from IPFS: ${err}`)
```

Comment:
not strictly related to the rest of this PR, but while debugging test failures I got frustrated at the very opaque `context deadline exceeded` error message and thought it would be nice to some more context to the error message.